src/Ssl/Generator/EcKey/EcKeyGenerator.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Acme PHP project.
+ *
+ * (c) Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AcmePhp\Ssl\Generator\EcKey;
+
+use AcmePhp\Ssl\Generator\KeyOption;
+use AcmePhp\Ssl\Generator\OpensslPrivateKeyGeneratorTrait;
+use AcmePhp\Ssl\Generator\PrivateKeyGeneratorInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * Generate random EC private key using OpenSSL.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class EcKeyGenerator implements PrivateKeyGeneratorInterface
+{
+    use OpensslPrivateKeyGeneratorTrait;
+
+    /**
+     * @param EcKeyOption|KeyOption $keyOption
+     */
+    public function generatePrivateKey(KeyOption $keyOption)
+    {
+        Assert::isInstanceOf($keyOption, EcKeyOption::class);
+
+        return $this->generatePrivateKeyFromOpensslOptions(
+            [
+                'private_key_type' => OPENSSL_KEYTYPE_EC,
```

Comment:
You might consider adding a `defined('OPENSSL_KEYTYPE_EC')` here and throw an exception if it is `false`. Not all OpenSSL versions support this key type.

Reply:
Documentation says http://php.net/manual/en/openssl.key-types.php
> This constant was added in 5.2.0.

are you sure that some version does expose the constant?

Reply:
nvm. Imissed the note
> This constant is only available when PHP is compiled with OpenSSL 0.9.8+.

Reply:
could you please have a look to my last commit? maybe @tgalopin  too